src/main/java/seedu/address/logic/commands/schedule/ImportCommand.java

```diff
@@ -0,0 +1,132 @@
+    public static final String INVALID_FILE_MESSAGE =
+            "File is empty, does not exist, or is not in json format! %s's schedule is unchanged.";
+    public static final String FAILED_TO_READ_FILE_MESSAGE = "Failed to read from file: %s";
+    public static final String VALID_DATA_FORMAT_EXAMPLE = "{\n"
+                    + "  \"events\" : [ {\n"
+                    + "    \"eventDescription\" : \"CS2106 Tutorial\",\n"
+                    + "    \"date\" : \"2022-03-14\",\n"
+                    + "    \"time\" : \"10:00:00\",\n"
+                    + "    \"duration\" : \"PT1H\",\n"
+                    + "    \"recurFrequency\" : \"WEEKLY\"\n"
+                    + "  }, {\n"
+                    + "    \"eventDescription\" : \"Alice Birthday Surprise\",\n"
+                    + "    \"date\" : \"2022-03-14\",\n"
+                    + "    \"time\" : \"17:00:00\",\n"
+                    + "    \"duration\" : \"PT4H\",\n"
+                    + "    \"recurFrequency\" : \"WEEKLY\"\n"
+                    + "  } ]\n"
+                    + "}";
+    public static final String MESSAGE_FAILURE_WITH_EXAMPLE =
+            MESSAGE_FAILURE + "\nExample of valid data:\n" + VALID_DATA_FORMAT_EXAMPLE;
+
+    private final Index targetIndex;
+    private final Path filePath;
+
+    /**
+     * @param targetIndex of the person in the filtered person list to edit
+     * @param filePath of the file to be read
+     */
+    public ImportCommand(Index targetIndex, Path filePath) {
+        this.targetIndex = targetIndex;
+        this.filePath = filePath;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(targetIndex.getZeroBased());
+
+        Optional<JsonAdaptedSchedule> importedJsonAdaptedSchedule;
+        try {
+            importedJsonAdaptedSchedule =
+                    JsonUtil.readJsonFile(filePath, JsonAdaptedSchedule.class);
+        } catch (DataConversionException e) {
+            throw new CommandException(String.format(
+                    FAILED_TO_READ_FILE_MESSAGE + e.getMessage(), filePath.getFileName()));
+        }
+
+        if (!importedJsonAdaptedSchedule.isPresent()) {
+            throw new CommandException(String.format(INVALID_FILE_MESSAGE, personToEdit.getName()));
+        }
+        Schedule importedSchedule;
+        try {
+            importedSchedule = importedJsonAdaptedSchedule.get().toModelType();
+        } catch (IllegalValueException e) {
+            throw new CommandException(String.format(
+                    MESSAGE_FAILURE_WITH_EXAMPLE, filePath.getFileName(), personToEdit.getName()));
```

Comment:
Perhaps instead of showing the example in the tiny return result display, we can create an example file in data for them to edit from

Reply:
I dont think that would be good. Once the file becomes edited or accidentally deleted, the user wont know the correct format anymore.

Reply:
I think that editing from a template would make creating the schedule easier. Regarding user not knowing the correct format when edited, the example file should be produced every time they try to import an invalid schedule, replacing the existing file even if it were edited. It will make a worse experience if they are not able to see the whole format at once. Other than creating an example file, we should also include the proper format of creating a schedule in the UG, under a title called `Creating contact schedule`.

Reply:
If there were access rights that were not given to the app, then it would not be able to save the addressbook at all. The app will actually not be useful at all. (We should mention this is the UG)
In the `LogicManager`, addressbook is saved whenever a command is called. So that problem is unlike when the it has been working properly. In the `export` command, other than giving the feedback, I made an edit where I checked if the file already exist and create the file if it doesn't. we can do it here as well.

Reply:
I think we can create the example json file. In addition, we could add the correct form on UG and also mention it is not recommended for beginner user. We could add this feature to the advance user part on the UG. About the access right, I agree with Han Qin as the base code itself is creating a data/addressbook.json after each command is executed. So technically if this import got problem, everything (including the base code) will fail.